### PR TITLE
Fix Owner Validation

### LINKED ISSUE #3476: Setting an empty "owner" is allowed in the REST API
## Description
When updating the owner of an artifact, we are allowing an empty string.  That should not be a valid value and rejected by the server.

**Registry
Version**: 2.4.3
**Persistence type**: all

### Steps to Reproduce
1. Create a new artifact
2. Try changing the owner to "" (empty string)
3. It's allowed?!

This should not be allowed - it should fail with some sort of 4xx error, perhaps a `400`.  The error message should indicate that there is a missing parameter.

### PR DESCRIPTION
**Pull request by renjingxiao**
Hi @EricWittmann  and @carlesarnal ,

In this PR I have added checks for the `owner` to ensure it's neither `null` or an empty string. `MissingRequiredParameterException` will be thrown if the check fails.

It will close #3476 .

Thank you for reviewing.

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/rest/v2/GroupsResourceImpl.java [hunk 1]
```diff
@@ -259,7 +259,13 @@ public void updateArtifactOwner(String groupId, String artifactId, ArtifactOwner
         requireParameter("groupId", groupId);
         requireParameter("artifactId", artifactId);
 
-        ArtifactOwnerDto dto = new ArtifactOwnerDto(data.getOwner());
+        String owner = data.getOwner();
+        requireParameter("owner", owner);
```
Comment (carlesarnal) [inline app/src/main/java/io/apicurio/registry/rest/v2/GroupsResourceImpl.java:263]: Is calling the `requireParameter` method not sufficient for this purpose?

**Review by carlesarnal — CHANGES_REQUESTED** [review]
Please address the feedback or reply to the comment.

**Comment by renjingxiao** [comment]
Hi @carlesarnal , thank you for the review. I thought `requireParameter` only checks for `null`, so I added lines 264-266 to check if it's an empty value. If they are redundant I'll remove them.

**Comment by renjingxiao** [comment]
Hi @carlesarnal , 
I have added the unit tests. Could you please have a look? 
Thank you

**Review by EricWittmann — APPROVED** [review]
Looks perfect, thanks @renjingxiao

**Pull request merged**